packages/transaction-manager/src/storage.ts

```diff
@@ -0,0 +1,32 @@
+export type Promisable<T> = PromiseLike<T> | T;
+
+/**
+ * key-value storage
+ */
+export interface Storage<Schema> {
+  hasItem<K extends keyof Schema>(key: K): Promisable<boolean>;
+  getItem<K extends keyof Schema>(key: K): Promisable<Schema[K] | undefined>;
+  removeItem<K extends keyof Schema>(key: K): Promisable<boolean>;
```

Comment:
`void` here to keep the API similar with `localStorage`, which makes user easy to use it, and a low-level API should be simple when possible

```ts
{
  storage: localStorage
}
```